src/javascript/_common/auth.js

```diff
@@ -0,0 +1,117 @@
+const {
+    AppIDConstants,
+    LocalStorageConstants,
+    LocalStorageUtils,
+    URLConstants,
+    WebSocketUtils,
+} = require('@deriv-com/utils');
+const Analytics = require('./analytics');
+
+export const DEFAULT_OAUTH_LOGOUT_URL = 'https://oauth.deriv.com/oauth2/sessions/logout';
+
+export const DEFAULT_OAUTH_ORIGIN_URL = 'https://oauth.deriv.com';
+
+const SocketURL = {
+    [URLConstants.derivP2pProduction]: 'blue.derivws.com',
+    [URLConstants.derivP2pStaging]   : 'red.derivws.com',
+};
+
+export const getServerInfo = () => {
+    const origin = window.location.origin;
+    const hostname = window.location.hostname;
+
+    const existingAppId = LocalStorageUtils.getValue(LocalStorageConstants.configAppId);
+    const existingServerUrl = LocalStorageUtils.getValue(LocalStorageConstants.configServerURL);
+    // since we don't have official app_id for staging,
+    // we will use the red server with app_id=62019 for the staging-p2p.deriv.com for now
+    // to fix the login issue
+    if (origin === URLConstants.derivP2pStaging && (!existingAppId || !existingServerUrl)) {
+        LocalStorageUtils.setValue(LocalStorageConstants.configServerURL, SocketURL[origin]);
+        LocalStorageUtils.setValue(LocalStorageConstants.configAppId, `${AppIDConstants.domainAppId[hostname]}`);
+    }
+
+    const serverUrl = LocalStorageUtils.getValue(LocalStorageConstants.configServerURL) || localStorage.getItem('config.server_url') || 'oauth.deriv.com';
+
+    const appId = LocalStorageUtils.getValue(LocalStorageConstants.configAppId);
+    const lang = LocalStorageUtils.getValue(LocalStorageConstants.i18nLanguage);
+
+    return {
+        appId,
+        lang,
+        serverUrl,
+    };
+};
+
+export const getOAuthLogoutUrl = () => {
+    const { appId, serverUrl } = getServerInfo();
+
+    const oauthUrl = appId && serverUrl ? `https://${serverUrl}/oauth2/sessions/logout` : DEFAULT_OAUTH_LOGOUT_URL;
+
+    return oauthUrl;
+};
+
+export const getOAuthOrigin = () => {
+    const { appId, serverUrl } = getServerInfo();
+
+    const oauthUrl = appId && serverUrl ? `https://${serverUrl}` : DEFAULT_OAUTH_ORIGIN_URL;
```

Comment:
[Q] whats appId check used here for?

Reply:
Its just an extra check to ensure we have the app ID and server URL before returning the origin url